modules/unsupported/mongodb/README.rst

```diff
@@ -2,17 +2,128 @@ MongoDB DataStore
+
+* The following Java equivalents of BSON types are valid: String, Double, Long, Integer, Boolean, Date.
+
+For the following GeoJSON feature::
+
+   {   "type": "Feature",
+       "geometry": {
+           "type": "Point",
+           "coordinates": [ 45.52, -122.681944 ]
+       },
+       "properties": {
+           "city": "Portland",
+           "year": "2014"
+           "attendance": "840"
+       }
+   }
+
+Is described using the following schema::
+
+   {
+       "typeName": "places",
+       "userData": {
+           "collection": "places"
+        },
+       "geometryDescriptor": {
+           "localName": "location",
+           "crs": {
+               "properties": {
+                   "name": "urn:ogc:def:crs:EPSG:4326"
+               },
+               "type": "name"
+           }
+       },
+       "attributeDescriptors": [
+           {
+               "localName": "location",
+               "type": {
+                   "binding": "com.vividsolutions.jts.geom.Point"
+               },
+               "userData": {
+                   "encoding": "GeoJSON",
+                   "mapping": "geometry"
+               }
+           },
+           {   "localName": "city",
+               "type": { "binding": "java.lang.String" },
+               "userData": { "mapping": "properties.name" }
+           },
+           {   "localName": "year",
+               "type": { "binding": "java.lang.String" },
+               "userData": { "mapping": "properties.year" }
+           },
+           {   "localName": "attendance",
+               "type": { "binding": "java.lang.String" },
+               "userData": {  "mapping": "properties.attendance" }
+           }
+       ]
+   }
+   
+   
+File URI schema stores:
+
+* For the directory-based schema store, edit the JSON document with the typeName requiring modification.
+  
+  Schemas are written createSchema() without indenting, but you can indent the resulting file for readability if desired.
+
+MongoDB URI schema stores:
+
+* Using a MongoDB document manipulation tool, update or insert the schema document in the collection maintaining the document in a form that follows the JSON schema format.
+  
+The JSON files contained in the file schema store are in a format that can be inserted into a MongoDB schema store (as long as the typeName in the file is unique to the document collection ).
+
+Multiple schemas, or views, can be created for a single MongoDB document collection by creating a new, unique, typeName and specifying the collection under the root-level userData object.
+
+Implementation Notes
+--------------------
+
+* Bounding box calculation makes use of a full table scan.
+
+* Multigeometry support requires MongoDB versions 2.5 and newer
+
+* Self-intersecting polygons is a common data problem preventing MongoDBDataStore from functioning.
```

Comment:
This might need some clarification to ease debugging of problematic datasets.  As MongoDB relies on data in WGS84 ingested data (e.g. through the importer) need to be transformed to WGS84.  While self-intersecting polygons may not exist in the original data in its native CRS, self-intersection may be introduced with a transform to WGS84.  This was observed early in testing against shapefiles representing detailed polygons in HARN projections.